web/backend/src/Server.ts

```diff
@@ -216,10 +216,23 @@ app.post('/api/add_role', (req, res) => {
   const { sciper } = req.body;
   const { role } = req.body;
 
-  usersDB.put(sciper, role).catch((error) => {
-    res.status(500).send('Failed to add role');
-    console.log(error);
-  });
+  // The sciper has to contain 6 numbers
+  if (sciper < 999999 && sciper > 100000) {
+    // call https://search-api.epfl.ch/api/ldap?q=228271 if the answer
+    // empty then sciper invalid
+    axios.get(`https://search-api.epfl.ch/api/ldap?q=${sciper}`).then((response) => {
+      if (response.data.length === 0) {
+        res.status(400).send('Unknown sciper');
+      } else {
+        usersDB.put(sciper, role).catch((error) => {
+          res.status(500).send('Failed to add role');
+          console.log(error);
+        });
+      }
+    });
```

Comment:
For you information, you can also concatenate promises, by returning a promise from within the body of a `then`.

This is sometimes more readable, and you can group all the errors in one catch.
```suggestion
    axios.get(`https://search-api.epfl.ch/api/ldap?q=${sciper}`)
      .then((response) => {
        if (response.data.length === 0) {
          res.status(400).send('Unknown sciper');
          return;
        }
        
        return usersDB.put(sciper, role);
      })
      .catch((error) => {
        res.status(500).send('Failed to add role');
        console.log(error);
      });;
```